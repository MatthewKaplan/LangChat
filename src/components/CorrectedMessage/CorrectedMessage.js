import React from 'react';
import { View, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

const CorrectedMessage = props => {
	return (
		<View>
			<Text>
				<Ionicons name="md-close" size={20} />
				{props.reference.message}
			</Text>
			<Text>
				<Ionicons name="ios-checkmark" size={20}/>
				{props.message}
			</Text>
		</View>
	);
};

export default CorrectedMessage;
