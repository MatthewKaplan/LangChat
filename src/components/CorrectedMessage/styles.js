import { StyleSheet } from 'react-native';

const lightgrey = '#d4d4d4';

export default StyleSheet.create({
	quote: {
		marginTop: 5,
		marginBottom: 5,
		padding: 5,
		paddingLeft: 10,
		borderColor: lightgrey,
		borderWidth: 1,
		borderRadius: 3
	},
	correction: {
		flexDirection: 'row',
		justifyContent: 'flex-start',
		alignItems: 'center',
		width: '95%'
	},
	correctionText: {
		marginLeft: 10
	}
});
