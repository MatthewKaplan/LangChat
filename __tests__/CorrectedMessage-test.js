// __tests__/CorrectedMessage-test.js
import React from 'react';
import CorrectedMessage from '../src/components/CorrectedMessage/CorrectedMessage';

import renderer from 'react-test-renderer';

const mockReference = {
	id: 89,
	message: 'Got a spot for us to roll',
	reference: null,
	room: 2,
	timestamp: '2019-07-22T20:34:02.045Z',
	user: 2,
	username: 'test_user1'
};

describe('CorrectedMessage Component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<CorrectedMessage reference={mockReference} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
