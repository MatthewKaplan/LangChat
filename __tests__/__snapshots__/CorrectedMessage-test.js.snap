// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorrectedMessage Component should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "borderColor": "#d4d4d4",
        "borderRadius": 3,
        "borderWidth": 1,
        "marginBottom": 5,
        "marginTop": 5,
        "padding": 5,
        "paddingLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontStyle": "italic",
          "marginBottom": 5,
        }
      }
    >
      test_user1
       said:
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        Got a spot for us to roll
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingLeft": 10,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
  </View>
</View>
`;
